core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1468,6 +1514,45 @@ public Builder setPreviousFileLocation(String previousFileLocation) {
       return this;
     }
 
+    private Builder setRowLineage(Boolean newRowLineage) {
+      if (newRowLineage == null) {
+        return this;
+      }
+
+      boolean disablingRowLineage = rowLineage && !newRowLineage;
+
+      Preconditions.checkArgument(
+          !disablingRowLineage, "Cannot disable row lineage once it has been enabled");
+
+      if (!rowLineage && newRowLineage) {
+        return enableRowLineage();
+      } else {
+        return this;
+      }
+    }
+
+    public Builder enableRowLineage() {
```

Comment:
it feels a bit weird to have both methods to effectively do the same. Maybe we should just reduce this to having a single method?

Reply:
Oops sorry, when I meant to do here is we only allow a public method that turns on RowLineage. There is no public way to turn off table lineage. So I think this is good as is.